### ISSUE TITLE
keep getting SIGSEGV: Segmentation Fault crash

### ISSUE THREAD

**Issue by justsignedupagain** [CleverRaven/Cataclysm-DDA#78772]
### Describe the bug

the past few hours, this save has been getting more and more frequent crashes, always the same error message (see screenshot) and now i can't move more than a few tiles before it crashes.

### Attach save file

[Stowell.zip](https://github.com/user-attachments/files/18251638/Stowell.zip)

### Steps to reproduce

1. open the save
2. walk around

### Expected behavior

i expect the game to not crash when i try to move

### Screenshots

![image](https://github.com/user-attachments/assets/683f31ab-fc44-436b-8143-d59a674febf8)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22631.4602 (23H2)
- Game Version: 7ba4c09 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context

[crash.log](https://github.com/user-attachments/files/18251989/crash.log)
[debug.log](https://github.com/user-attachments/files/18251990/debug.log)

**Comment by RenechCDDA** [comment]
Please attach a save archive as the form instructs.

**Comment by PatrikLundell** [comment]
Failed to repeat.
I loaded the save, ignored a tough zombie and walked east until encountering zombies, ignored, walked back until warned about the tough zombie again, and ran back and forth that way until the tough zombie eventually caught up due to darkness causing the PC to get close enough to it for it to detect the PC and give chase.

**Comment by RenechCDDA** [comment]
> Failed to repeat. I loaded the save, ignored a tough zombie and walked east until encountering zombies, ignored, walked back until warned about the tough zombie again, and ran back and forth that way until the tough zombie eventually caught up due to darkness causing the PC to get close enough to it for it to detect the PC and give chase.

The stack trace suggests it is occurring as a result of a monster stepping onto a trap and teleporting(???). Without the actual local maps from their save file reproducing it will require guesswork and setup.

**Comment by PatrikLundell** [comment]
I was confused by the teleporting part, but a teleport trap makes sense.
There's a flesh overgrown lab beneath, and labs have a tendency to contain teleport traps.

**Comment by inogenous** [comment]
Issue can be reproduced by walking south after loading the save.

Possible cause is as follows:
1. Some monster walks into a teleport trap, and the selected teleport destination is out of bounds for the current game map
2. `teleport::teleport_to_point` handles out of bounds by creating a temporary map (see line 82): https://github.com/CleverRaven/Cataclysm-DDA/blob/163ee186031763cf0ed5c5299362963121d97fa4/src/teleport.cpp#L82-L94
3. `teleport::teleport_to_point` also calls `dest->spawn_monsters()` on this temporary map (line 91 above). Please note that this map is not the same as the main game map. Bub coordinates in this temporary map are therefore not equivalent to bub coordinates on the main game map.
4. In `map::spawn_monsters_submap`, we calculate local bub coords for where to spawn monsters. But as stated above, note that these bub coords are relative to this temporary map, and not the same as the game map.
5. `map::spawn_monsters_submap` tries to place monsters by calling `g->place_critter_at( p )` using a local bub coord that is relative to the temporary map from step 2 :  <-- this is the problem  https://github.com/CleverRaven/Cataclysm-DDA/blob/163ee186031763cf0ed5c5299362963121d97fa4/src/map.cpp#L9455-L9463
6. The monster could not be placed, and the game segfaults when doing line 9458 above because `placed` is null.

**Comment by harakka** [comment]
Should be fixed by #79053

**Comment by PatrikLundell** [comment]
Not really. That PR removes the symptoms so the game doesn't crash, but it doesn't deal with the underlying cause. That's why #79081 is marked as closing this bug report. Not a big deal, though, since there are two bug reports that have been found to be about this issue, and the PR exists regardless.

**Comment by harakka** [comment]
Thanks for clarifying, I misunderstood the PR text then. I'll reopen so it'll be tracked properly.
